packages/go_router/lib/src/configuration.dart

```diff
@@ -264,28 +265,26 @@ class RouteConfiguration {
           '${queryParameters.isEmpty ? '' : ', queryParameters: $queryParameters'}');
       return true;
     }());
-    assert(_nameToPath.containsKey(name), 'unknown route name: $name');
-    final String path = _nameToPath[name]!;
+    assert(_routesByName.containsKey(name), 'unknown route name: $name');
+    final GoRoute route = _routesByName[name]!;
     assert(() {
-      // Check that all required params are present
-      final List<String> paramNames = <String>[];
-      patternToRegExp(path, paramNames);
-      for (final String paramName in paramNames) {
+      for (final String paramName in route.pattern.parameters) {
         assert(pathParameters.containsKey(paramName),
-            'missing param "$paramName" for $path');
+            'missing param "$paramName" for ${route.pattern}');
```

Comment:
`RoutePattern` overrides toString

Reply:
this may be a breaking change, this could be an easy migration if you write rules fix_data.yaml

Also in this library, `path` means `/book/:bookId`, the filled string like `book/123` is called `location`.